src/couch_replicator_manager.erl

```diff
@@ -823,3 +825,38 @@ is_replicator_db(Name) ->
         _ ->
             false
     end.
+
+possibly_hack(Rep, DbName) ->
+    #rep{
+        source = Src,
+        target = Tgt
+    } = Rep,
+    Rep#rep{
+        source = fix_uri(DbName, Src),
+        target = fix_uri(DbName, Tgt)
+    }.
+
+fix_uri(_ReplicationDbName, #httpdb{} = Db) ->
+    Db;
+fix_uri(ReplicationDbName, DbName) ->
+    case customer_name(ReplicationDbName) of
+        undefined ->
+            DbName;
+        Customer ->
+            case customer_name(DbName) of
```

Comment:
I think we should eliminate this case. There is not usable case for customers writing to node local dbs. In fact they won't even be able to create those dbs as they require server_admin credentials.

To simplify the check we could just explicitly check that DbName is <<"_replicator">> instead of bothering parsing out the customer. Also in this replicator version `<<"shards">>, _, Customer` would be a valid case?